Cabal-tests/tests/HackageTests.hs

```diff
@@ -63,7 +63,7 @@ import Data.TreeDiff.Pretty          (ansiWlEditExprCompact)
 parseIndex :: (Monoid a, NFData a) => (FilePath -> Bool)
            -> (FilePath -> B.ByteString -> IO a) -> IO a
 parseIndex predicate action = do
-    cabalDir   <- getAppUserDataDirectory "cabal"
+    cabalDir   <- getXdgDirectory XdgData "cabal"
```

Comment:
This is also not right. E.g. `~/.cabal/config` is `XdgConfig`, and about everything else (e.g. `~/.cabal/store`, `~/.cabal/packages`) are probably `XdgCache`.

What about `~/.cabal/bin` ?

Reply:
Yeah, I wasn't sure either of how cabal was processing all of its config, so I figured the fastest way to see was to draft up a PR for review. Will change the appropriate paths once validate finishes to complete.

`~/.cabal/bin` should be `~/.local/bin` as per the systemd file system specification, no user variable there. This one is a little bit less thought out but that's the current take on it.